src/details/ArborX_DetailsTreeConstruction.hpp

```diff
@@ -213,6 +213,25 @@ inline void TreeConstruction<DeviceType>::calculateBoundingBoxOfTheScene(
   Kokkos::fence();
 }
 
+template <typename Primitives>
+struct GeometryTagHelper
+{
+private:
+  using accessor_return_type = std::decay_t<decltype(
+      Traits::Access<Primitives, Traits::PrimitivesTag>::get(
+          std::declval<Primitives const &>(), std::declval<int>()))>;
```

Comment:
So this looks good but I wonder of the implications. Does that mean that we won't respect `cv` qualifiers?

Reply:
No it means that it is up to the user if he wants to store ArborX geometric objects and return references or generate them on the fly.
We don't care about cv qualifiers, all we need to know is was it a point or a box.